src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -1038,13 +1039,18 @@ public function offsetGet($option, bool $triggerDeprecation = true)
 
             foreach ($this->allowedValues[$option] as $allowedValue) {
                 if ($allowedValue instanceof \Closure) {
-                    if ($allowedValue($value)) {
+                    try {
+                        if (!$allowedValue($value)) {
+                            // Don't include closures in the exception message
+                            continue;
+                        }
+
                         $success = true;
                         break;
+                    } catch (ValidationFailedException $e) {
+                        // Don't include closures in the exception message
```

Comment:
If you keep the `if (!$allowedValue($value)) {` condition, you could avoid duplicating the comment and the `continue`, by keeping it after the try/catch.